cluster/terraform/main.tf

```diff
@@ -61,20 +62,55 @@ resource "azurerm_kubernetes_cluster" "main" {
   lifecycle { ignore_changes = [tags] }
 }
 
-resource "azurerm_private_dns_zone" "backing-services-dns-zones" {
-  for_each = local.subnets
+resource "azurerm_kubernetes_cluster_node_pool" "node_pools" {
+  for_each = var.node_pools
+
+  name                  = each.key
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.main.id
+  vm_size               = each.value.vm_size
+  enable_auto_scaling   = each.value.enable_auto_scaling
+  node_count            = each.value.node_count
+  min_count             = each.value.min_count
+  max_count             = each.value.max_count
+  node_labels           = each.value.node_labels
```

Comment:
Could it be set to something like `"node_labels": {"node_pool": "${each.key}"},` ?
Similarly for node_taints